src/app/utils/field-validation/validators/dateValidator.ts

```diff
@@ -0,0 +1,51 @@
+import { chain, left, right } from 'fp-ts/lib/Either'
+import { flow } from 'fp-ts/lib/function'
+import moment from 'moment-timezone'
+
+import { IDateField } from 'src/types/field'
+import { ResponseValidator } from 'src/types/field/utils/validation'
+import { ISingleAnswerResponse } from 'src/types/response'
+
+import { createMomentFromDateString } from '../../../../shared/util/date-validation'
+
+import { notEmptySingleAnswerResponse } from './common'
+
+type DateValidator = ResponseValidator<ISingleAnswerResponse>
+type DateValidatorConstructor = (dateField: IDateField) => DateValidator
+
+const dateFormatValidator: DateValidator = (response) => {
+  const { answer } = response
+  return createMomentFromDateString(answer).isValid()
+    ? right(response)
+    : left(`DateValidator:\t answer is not a valid date`)
+}
+
+const makeDateValidator: DateValidatorConstructor = (dateField) => (
+  response,
+) => {
+  const { answer } = response
+  const answerDate = createMomentFromDateString(answer)
+  const { selectedDateValidation, customMinDate, customMaxDate } =
+    dateField.dateValidation || {}
+
+  const isFutureOnly = selectedDateValidation === 'Disallow past dates'
```

Comment:
Use the enum `DateSelectedValidation` instead of hardcoded string? So this won't erroneously pass even if the enum has changed; but then the frontend is not using that enum...........

Can you also create an issue to track this? Move the enum to `shared` and make frontend also rely on that enum instead of `DATE_VALIDATION_OPTIONS` in `src/public/modules/forms/admin/controllers/edit-fields-modal.client.controller.js`

Reply:
edited + created issue